### PR TITLE
Issue/14128 reader related posts UI refactoring

### PR DESCRIPTION
**Pull request by ashiagr**
Task #14128

This PR is a sub-task of Issue #14128  and refactors/ restructures related posts UI logic. It 

1. Builds `RelatedPostsUiState` in `ReaderPostDetailsUiStateBuilder` for the related posts. 
2. Displays related posts using a recycler view.

To test:
1. Open a wp.com reader post. 
2. Notice that local related posts (from selected post's site) are shown at the bottom of the post as recycler view cards with a featured image and title.
3. Notice that global related posts (from another wp.com site) are shown at the bottom of the post as recycler view cards with  featured image, title and follow button.

Tests included in `ReaderPostDetailsUiStateBuilderTest`.

Notes:

1. Related posts actions are implemented in #14163.
2. Ignore all stying. UI fine-tuning + cleanup (related to `ReaderSimplePost` + `ReaderSimplePostView`) will be done separately. 
3. Yet to set featured image size.

UI Design:

Waiting for 
- Follow button positioning for global posts.
- Whether to show just the title and no image if featured image is not set for a post. I haven't removed redundant fields from  `ReaderSimplePost` until design is finalised (in case we decide to show some other fields).

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/reader/viewmodels/ReaderPostDetailViewModel.kt [hunk 1]
```diff
@@ -203,7 +205,7 @@ class ReaderPostDetailViewModel @Inject constructor(
                 is FetchRelatedPostsState.Failed.NoNetwork,
                 is FetchRelatedPostsState.Failed.RequestFailed -> Unit // Do Nothing
```
Comment (zwarm) [inline WordPress/src/main/java/org/wordpress/android/ui/reader/viewmodels/ReaderPostDetailViewModel.kt]: Nitpick: Next time you're in the file want to remove the empty line, probably wasn't intentional.

**Review by zwarm — APPROVED** [review]
Nice work @ashiagr , it's looking good. Both global (wp.com) and local (same site) related posts are shown as cards, with global having the follow button visible. Happy to see more Kotlin and less Java :)

File: WordPress/src/main/java/org/wordpress/android/ui/reader/viewmodels/ReaderPostDetailViewModel.kt (hunk 1 above)
Comment (ashiagr) [reply WordPress/src/main/java/org/wordpress/android/ui/reader/viewmodels/ReaderPostDetailViewModel.kt]: Removed in 2d46f1e

**Comment by ashiagr** [comment]
Thanks for the review @zwarm!

I'll address these in "Fine-tune UI + cleanup" task based on final design that we'll get:

- Follow button positioning for global posts 
- Whether to show just the title and no image if featured image is not set for a post. I haven't removed redundant fields from ReaderSimplePost until design is finalised (in case we decide to show some other fields).

Merging this one.

**Pull request merged**